iocore/net/SSLUtils.cc

```diff
@@ -1723,6 +1749,64 @@ SSLReadBuffer(SSL *ssl, void *buf, int64_t nbytes, int64_t &nread)
     return SSL_ERROR_NONE;
   }
   ERR_clear_error();
+
+#if TS_HAS_TLS_EARLY_DATA
+  if (SSL_version(ssl) >= TLS1_3_VERSION) {
```

Comment:
Do we need this check here? `netvc->early_data_reader` should be `nullptr` if early data is not available. I'd add `bool is_early_data_available()` to SSLNetVC. IMO, the function should be defined in EarlyDataSupport class like ALPNSupport class because QUICNetVC should also support it, but it might be too much at the moment.

Reply:
We actually need this check here, the `nullptr` check isn't sufficient to avoid this situation:
enabled early data in config, but the actual tls version used is < 1.3

Reply:
It didn't before the latest push because the reader was allocated inside the check for TLS version :)

I still think we can remove this check. For example, by resetting `nullptr` after checking `netvc->early_data_reader->read_avail() == 0` in `SSLAccept`.

I think what we really need to check here is not TLS version but whether early data is available.

Reply:
yeah, but it was breaking before the last patch. :D 
I'll think about that.